src/components/Balance.test.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { render, screen } from '../testUtils'
+import { BTC, SATS } from '../utils'
+
+import Balance from './Balance'
+
+describe('<Balance />', () => {
```

Comment:
Do you think it would be good to add tests for `0`, `1`, [`2099999997690000`](https://en.bitcoin.it/wiki/Controlled_supply#Projected_Bitcoins_Long_Term), `2100000000000000` sats too?
Thinking of potential floating point issues - JM needs some amounts denominated in `BTC`, we should be confident to handle these well -

Reply:
I updated the value parsing and tests a bit, please have another look and let me know if it is at least a bit clearer now. Still not perfect but I hope that what is happening is at least a bit more transparent now.